cdap-app-fabric/src/test/java/co/cask/cdap/metadata/LineageAdminTest.java

```diff
@@ -494,6 +500,125 @@ public void testBranchLoopLineage() throws Exception {
     );
   }
 
+
+  @Test
+  public void testWorkflowLineage() throws Exception {
+    // Lineage for D3 -> P2 -> D2 -> P1 -> D1
+
+    LineageStore lineageStore = new LineageStore(getTxExecFactory(), getDatasetFramework(),
+                                                 NamespaceId.DEFAULT.dataset("testWorkflowLineage"));
+    Store store = getInjector().getInstance(Store.class);
+    MetadataStore metadataStore = getInjector().getInstance(MetadataStore.class);
+    LineageAdmin lineageAdmin = new LineageAdmin(lineageStore, store, metadataStore, new NoOpEntityExistenceVerifier());
+
+    // Define metadata
+    MetadataRecord run1AppMeta = new MetadataRecord(program1.getParent(), MetadataScope.USER,
+                                                    toMap("pk1", "pk1"), toSet("pt1"));
+    MetadataRecord run1ProgramMeta = new MetadataRecord(program1, MetadataScope.USER,
+                                                        toMap("pk1", "pk1"), toSet("pt1"));
+    MetadataRecord run1Data1Meta = new MetadataRecord(dataset1, MetadataScope.USER,
+                                                      toMap("dk1", "dk1"), toSet("dt1"));
+    MetadataRecord run1Data2Meta = new MetadataRecord(dataset2, MetadataScope.USER,
+                                                      toMap("dk2", "dk2"), toSet("dt2"));
+
+    // Add metadata
+    metadataStore.setProperties(MetadataScope.USER, program1.getParent(), run1AppMeta.getProperties());
+    //noinspection ToArrayCallWithZeroLengthArrayArgument
+    metadataStore.addTags(MetadataScope.USER, program1.getParent(), run1AppMeta.getTags().toArray(new String[0]));
+    metadataStore.setProperties(MetadataScope.USER, program1, run1ProgramMeta.getProperties());
+    //noinspection ToArrayCallWithZeroLengthArrayArgument
+    metadataStore.addTags(MetadataScope.USER, program1, run1ProgramMeta.getTags().toArray(new String[0]));
+    metadataStore.setProperties(MetadataScope.USER, dataset1, run1Data1Meta.getProperties());
+    //noinspection ToArrayCallWithZeroLengthArrayArgument
+    metadataStore.addTags(MetadataScope.USER, dataset1, run1Data1Meta.getTags().toArray(new String[0]));
+    metadataStore.setProperties(MetadataScope.USER, dataset2, run1Data2Meta.getProperties());
+    //noinspection ToArrayCallWithZeroLengthArrayArgument
+    metadataStore.addTags(MetadataScope.USER, dataset2, run1Data2Meta.getTags().toArray(new String[0]));
+    TimeUnit.MILLISECONDS.sleep(1);
```

Comment:
I don't think sleep is required here. I see that `testSimpleLineage` also  has a sleep, can you please remove that too?